coderd/userauth.go

```diff
@@ -435,10 +545,8 @@ func (api *API) oauthLogin(r *http.Request, params oauthLoginParams) (*http.Cook
 			err  error
 		)
 
-		user, link, err = findLinkedUser(ctx, tx, params.LinkedID, params.Email)
-		if err != nil {
-			return xerrors.Errorf("find linked user: %w", err)
-		}
+		user = params.User
+		link = params.Link
```

Comment:
I thought it was safe to lift finding the user outside of the transaction and into the callers so we can access `User.ID` in those places. If, in the callers, we don't find a user, we bail early, never entering `oauthLogin`.